nerdlandbot/scheduler/PurgeScheduler.py

```diff
@@ -14,17 +14,19 @@ async def purge_messages(bot):
     info("Purging messages")
     guilds_data = await get_all_guilds_data()
     for guild_data in guilds_data:
-        for text_channel, max_age in guild_data.purgers.items():
-            before = datetime.today() - timedelta(days=max_age)
-            channel = bot.get_channel(int(text_channel))
-            try:
-                if channel:
-                    await channel.purge(check=check, before=before)
-            except:
-                fatal(
-                    f"Failed to purge messages for channel {channel.name} in guild {guild_data.guild_id}. Does the bot have appropriate permissions on that channel?"
-                )
-
-
+        if bot.is_purging[str(guild_data.guild_id)] == False:
+            bot.is_purging[str(guild_data.guild_id)] = True
+            for text_channel, max_age in guild_data.purgers.items():
+                before = datetime.today() - timedelta(days=max_age)
+                channel = bot.get_channel(int(text_channel))
+                try:
+                    if channel:
+                        await channel.purge(check=check, before=before)
+                except:
+                    fatal(
```

Comment:
If I recall correctly, calling fatal will produce a log indicating that the process failed, but it will not crash the bot.
It will however end this method call, and the purging flag will be left on true.
I would reset the flag in a finally clause to make sure it is always executed